modules/tensor_mechanics/src/materials/ADComputeFiniteStrain.C

```diff
@@ -147,9 +147,6 @@ ADComputeFiniteStrain::computeQpIncrements(ADRankTwoTensor & total_strain_increm
       const ADReal C1_squared =
           p + 3.0 * Utility::pow<2>(p) * (1.0 - (p + q)) / Utility::pow<2>(p + q) -
           2.0 * Utility::pow<3>(p) * (1.0 - (p + q)) / Utility::pow<3>(p + q);
-      mooseAssert(C1_squared >= 0.0,
-                  "Cannot take square root of a negative number. This may happen when elements "
-                  "become heavily distorted.");
```

Comment:
We need a way to terminate this loop.

For example, if we have 100 elements, and we have this negative at the first element  and we know we should cut time step, but we have to go through all other 99 elements. 

Should we do that same thing for` mooseAssert(C3_test >= 0.0,` ?

Reply:
We don't know that we need to cut the time step. Cutting the lambda when doing a line search may be sufficient (and is indeed exactly what happens when running in non-debugging modes for the blackbear test)